app/code/Magento/Wishlist/Block/Customer/Wishlist.php

```diff
@@ -78,14 +83,66 @@ protected function _prepareCollection($collection)
     }
 
     /**
-     * Preparing global layout
+     * Paginate Wishlist Product Items collection
      *
      * @return void
      */
+    public function paginateCollection()
+    {
+        $page = $this->getRequest()->getParam("p", 1);
+        $limit = $this->getRequest()->getParam("limit", 10);
+        $this->_collection
+            ->setPageSize($limit)
+            ->setCurPage($page);
+    }
+
+    /**
+     * Retrieve Wishlist Product Items collection
+     *
+     * @return \Magento\Wishlist\Model\ResourceModel\Item\Collection
+     */
+    public function getWishlistItems()
+    {
+        if ($this->_collection === null) {
+            $this->_collection = $this->_createWishlistItemCollection();
+            $this->_prepareCollection($this->_collection);
+            $this->paginateCollection();
+        }
+
+        return $this->_collection;
+    }
+
+    /**
+     * Preparing global layout
+     *
+     * @return $this
+     */
     protected function _prepareLayout()
     {
+        $this->getWishlistItems();
         parent::_prepareLayout();
         $this->pageConfig->getTitle()->set(__('My Wish List'));
+        $this->getChildBlock('wishlist_item_pager')
+            ->setUseContainer(
+                true
+            )->setShowAmounts(
+                true
+            )->setFrameLength(
+                $this->_scopeConfig->getValue(
+                    'design/pagination/pagination_frame',
+                    \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+                )
+            )->setJump(
+                $this->_scopeConfig->getValue(
+                    'design/pagination/pagination_frame_skip',
+                    \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+                )
+            )->setLimit(
+                $this->getLimit()
+            )
+            ->setCollection($this->_collection);
+        $this->_collection->load();
```

Comment:
There is no need to load the collection before it is used, it will be loaded during template rendering

Reply:
done